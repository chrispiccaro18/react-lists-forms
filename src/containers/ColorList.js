import React, { PureComponent } from 'react';
import Colors from '../components/Colors';
import CreateColor from '../components/CreateColor';

export default class ColorList extends PureComponent {
  state = {
    colors: []
  }

  addColor = color => {
    this.setState(state => (
      { colors: [...state.colors, color] }
    ));
  }

  render() {
    const { colors } = this.state;
    return (
      <>
        <CreateColor addColor={this.addColor}/>
        <Colors colors={colors}/>
      </>
    );
  }
}
