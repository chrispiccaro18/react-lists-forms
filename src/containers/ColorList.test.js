import React from 'react';
import { shallow } from 'enzyme';
import ColorList from './ColorList';

describe('ColorList component', () => {
  it('renders a ColorList', () => {
    const wrapper = shallow(<ColorList />);
    expect(wrapper).toMatchSnapshot();
  });
});
