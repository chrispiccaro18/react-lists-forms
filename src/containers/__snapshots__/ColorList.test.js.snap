// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorList component renders a ColorList 1`] = `
<Fragment>
  <CreateColor
    addColor={[Function]}
  />
  <Colors
    colors={Array []}
  />
</Fragment>
`;
